src/redux/actions/concepts/addBulkConcepts/index.js

```diff
@@ -107,7 +107,13 @@ export const addConcept = (params, data, conceptName, id) => async (dispatch) =>
     );
     data.data.expressions.push(...referencesToAdd);
 
-    updateNotification('Finalizing...');
+    if (referencesToAdd.length) {
+      updateNotification(
+        `Adding this and ${referencesToAdd.length} dependent concepts...\nThis may take several minutes. DO NOT NAVIGATE AWAY FROM THE APPLICATION UNTIL THIS IS COMPLETE.`,
```

Comment:
Why are we duplicating this message?

Reply:
Sorry I missed this @dkayiwa.

Extracting the repeated part of this string to separate constant.

Reply:
Done.